### PR TITLE
chore: add metrics for unique users reporter

### PR DESCRIPTION
**Pull request by mihir20**
# Description

Add following metrics in rudder server

- Latency of generating HLLs

- Size in each HLL inserted in the outbox table

## Linear Ticket

https://linear.app/rudderstack/issue/PIPE-1245/add-metrics-for-rudder-server

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: processor/processor.go [hunk 1]
```diff
@@ -607,6 +608,11 @@ func (proc *Handle) Setup(
 			"partition": partition,
 		})
 	}
+	proc.stats.trackedUsersReportGeneration = func(partition string) stats.Measurement {
+		return proc.statsFactory.NewTaggedStat("processor_tracked_users_report_gen_time", stats.TimerType, stats.Tags{
```
Comment (lvrach) [inline processor/processor.go:612]: [minor] from prometheus naming best practises, postfix with the unit
```suggestion
		return proc.statsFactory.NewTaggedStat("processor_tracked_users_report_gen_seconds", stats.TimerType, stats.Tags{
```

File: runner/buckets.go [hunk 2]
```diff
@@ -108,5 +108,13 @@ var (
...
 			0.1, 0.5, 1, 5, 60, 300, 600, 1800, 3600, 12 * 3600, 24 * 3600,
 		},
+		"tracked_users_hll_bytes": {
+			float64(10 * bytesize.B),  // for hll containing single id = 8B
+			float64(100 * bytesize.B), // for hll containing 10 ids = 80B
+			float64(1 * bytesize.KB),  // for hll containing 100 ids = 800B
+			float64(3 * bytesize.KB),  // for hll containing 300 ids = 2400B
+			float64(6 * bytesize.KB),  // for hll containing 700 ids = 5600B
+			float64(10 * bytesize.KB), // max size for hll with log2m=14, regWidth=5 = 10KB
+		},
```
Comment (lvrach) [inline runner/buckets.go:118]: Will need to fine tune the buckets for this one I believe
```suggestion
		},
		"processor_tracked_users_report_gen_seconds": {
		     // 1ms, 10ms, 50ms, 100ms, 0.5s, 1s
		}
```

Comment (mihir20) [reply runner/buckets.go:118]: yes this is in micro seconds, I will update this.

Comment (mihir20) [reply runner/buckets.go:124]: For now we can keep this metric for some time, we can decide and remove later after checking in prod. May be in next server release?

File: runner/buckets.go [hunk 3]
```diff
@@ -108,5 +108,19 @@ var (
...
 		},
+		"tracked_users_hll_bytes": {
+			float64(10 * bytesize.B),  // for hll containing single id = 8B
+			float64(100 * bytesize.B), // for hll containing 10 ids = 80B
+			float64(400 * bytesize.B), // for hll containing 50 ids = 400B
+			float64(1 * bytesize.KB),  // for hll containing 100 ids = 800B
+			float64(3 * bytesize.KB),  // for hll containing 300 ids = 2400B
+			float64(6 * bytesize.KB),  // for hll containing 700 ids = 5600B
+			float64(10 * bytesize.KB), // max size for hll with log2m=14, regWidth=5 = 10KB
+			float64(41 * bytesize.KB), // max size for hll with log2m=16, regWidth=5 = 40KB
```
Comment (satishrudderstack) [inline runner/buckets.go:119]: @mihir20 Are we able to pick these buckets for the metric. In the load test I remember it was not using these buckets ?

**Review by satishrudderstack — APPROVED** [review]
LGTM

File: runner/buckets.go (hunk 3 above)
Comment (mihir20) [reply runner/buckets.go:119]: In load test we were using statsd for metrics which do not support custom buckets, opentelemetry supports custom bucket but we metrics were not flowing when we enabled otel in devcluster.

**Review by lvrach — APPROVED** [review]

**Pull request merged**